content/dev/blog/analyzing-so-data.Rmd

```diff
@@ -0,0 +1,308 @@
+
+```{r cache = TRUE}
+ds$compensation_MR <- makeMR(ds[, names(mr_vars)],
+    selections = "selected", name = "compensation_MR")
+knitr::include_graphics('./images/compensation_MR_screen.png')
+```
+
+
+### Correcting misreported salaries
+
+Next we're going to replicate and extend a [procedure used by Evelina
+Gabasova](http://evelinag.com/blog/2017/06-20-stackoverflow-tabs-spaces-and-salary/#.We4QuxNSxTY)
+to correct for misreported salary data from Eastern Europe. Evelina noticed that
+there were discrepancies in the distributions of salary data. For some
+countries, even within countries, there was a multi-modal distribution of
+salaries that she was surprised by. In fact, for a number of countries, it
+looked like there were two distributions, one centered at approximately 1/12 of
+the other. This lead her to wonder if maybe some developers from countries where
+salaries are frequently given by month had actually reported their monthly
+rather than yearly salaries.
+
+Left alone, this would skew the salaries for the countries where this is common
+much lower than reality. Inspired by Evelina's approach of using mixture models,
+which she fit to the data and then corrected the low-salary group by multiplying
+by 12 to get an annualized salary, we used mixture models and a some heuristics
+to determine which countries had suspect distributions of salaries, and then
+corrected those that fit this monthly-annual discrepancy. Like Evelina, we fit
+mixture models using the [mclust
+package](http://www.stat.washington.edu/mclust/). We fit mixture models for
+every country, and then adjusted salaries under the following conditions:
+
+* if there is only one mode / cluster don't adjust the salaries. This is the simplest
+case, if there's no discrepancy there's no reason to correct.
+* if any mode /cluster of salaries only represented 10% of the population, don't adjust the
+salaries. We found that sometimes `mclust` would find very small clusters and
+that would produce odd results, we chose 10% here out of convenience, but a more
+principle approach would pick a better cut off.
+* if there was more than one mode / cluster of salaries, and the lowest group's salary
+multiplied by 12 was greater than the second lowest group's salary multiplied by two, then do not
+attempt to adjust the salaries. Sometimes `mclust` would find clusters, but they
+would actually be fairly tightly together. This was likely the result of other
+covariates like experience, but critically, we didn't want to multiply junior
+developers' salaries by 12 just because they form a distinct cluster separate
+from senior developers.
+* otherwise, multiply the lowest salary group by 12 to correct for apparent misreporting.
+
+Finally, we only applied this adjustment to countries where we had at least 90
+responses. This allowed us to be reasonably confident that there were enough
+salaries in the sample to detect distinct clusters.
+
```

Comment:
This section should discuss how the crunch package lets you work with the data efficiently and only pull the data you need when you work on it. And we also have `crplyr` that provides a dplyr backend, which we'll use here.